dj/construction/compile.py

```diff
@@ -422,4 +414,5 @@ def compile_node(
     if node.query is None:
         raise DJException(f"Cannot compile node `{node.name}` with no query.")
     query = parse(node.query, dialect)
-    return compile_query_ast(session, query)
+    compile_query_ast(session, query)
```

Comment:
Would this fit as a method on the query class? Then this could be changed to `query.compile(session)`

Reply:
I'm even still skeptical of the `build` method on `Query`. The steps really all work in concert with one another I'm not sure what the use case is to really be calling them out of the blue.

Reply:
Of course either works but for me it's about the scope. From what I understand this function only works on query instances and manipulates them in place, the function name tries to make that clear but I think even clearer is a method on an instance that makes some change to the instance's state. Then even with a simple name like `compile`, it's clear that it doesn't operate on Column nodes, as an example.

Reply:
okay I added the compile method to `Query`, everywhere now just uses the `build` and `compile` methods and I got rid of the old functions and consolidated the little bit of logic specific to `Query` into the methods

Reply:
+to consolidate I stole the method to add aliases